### PR TITLE
feat(export/import configuration) : support namespace configurations

### PR DESCRIPTION
**Pull request by jonathanweyermann**
image

### CODE REVIEW

File: apps/tenant-management-webapp/src/app/pages/admin/services/configuration/export/export.tsx [hunk 1]
```diff
@@ -237,7 +259,7 @@ export const ConfigurationExport: FunctionComponent = () => {
                                       <GoAChip
                                         key={exp}
                                         deletable={true}
-                                        content={shortName}
+                                        content={shortName || 'all names'}
```
Comment (rstyan) [inline apps/tenant-management-webapp/src/app/pages/admin/services/configuration/export/export.tsx:262]: Use the constant for all names

File: apps/tenant-management-webapp/src/app/store/configuration/sagas.ts [hunk 2]
```diff
@@ -131,25 +140,38 @@ export function* fetchConfigurations(action: FetchConfigurationsAction): SagaIte
...
-              c.data['description'] = definitions[key]?.configurationSchema?.description;
-            }
-            return c.data;
-          })
+          configs
+            .map((c) => {
+              if (Array.isArray(c)) {
+                return c;
+              }
+              const key = `${c.data.namespace}:${c.data.name}`;
```
Comment (rstyan) [inline apps/tenant-management-webapp/src/app/store/configuration/sagas.ts:168]: I'm a bit lost here: don't we have to worry about the name===null here?  Shouldn't this use the (new) toServiceKey() function?

File: apps/tenant-management-webapp/src/app/store/configuration/sagas.ts [hunk 3]
```diff
@@ -131,25 +140,38 @@ export function* fetchConfigurations(action: FetchConfigurationsAction): SagaIte
...
-  const urls = getFetchUrls(action.services, configBaseUrl);
-  if (configBaseUrl && token && urls.length > 0) {
+
+  const urlsData = getFetchUrls(action.services, configBaseUrl);
+  if (configBaseUrl && token && urlsData.length > 0) {
     try {
       const configs = yield all(
-        urls.map((url) => call(axios.get, url, { headers: { Authorization: `Bearer ${token}` } }))
+        urlsData.map((urlData) => {
+          if (urlData.namespace) {
```
Comment (rstyan) [inline apps/tenant-management-webapp/src/app/store/configuration/sagas.ts:149]: I don't think this if/else is needed, see above comment.

File: apps/tenant-management-webapp/src/app/store/configuration/sagas.ts [hunk 4]
```diff
@@ -118,6 +104,29 @@ export function* fetchConfigurationDefinitions(_action: FetchConfigurationDefini
   }
 }
 
+async function fetchConfigurationNamespace(urlData: UrlData, token: string) {
```
Comment (rstyan) [inline apps/tenant-management-webapp/src/app/store/configuration/sagas.ts:107]: This should work with both mult-page calls and single page calls.  There shouldn't be any need to have the if statement on line 149.  If not, it shouldn't be hard to make it work for both.

**Review by solittlework — APPROVED** [review]

File: apps/tenant-management-webapp/src/app/store/configuration/sagas.ts [hunk 5]
```diff
@@ -118,6 +104,29 @@ export function* fetchConfigurationDefinitions(_action: FetchConfigurationDefini
...
+  let hasMoreData = true;
+
+  while (hasMoreData) {
+    let url = `${urlData.url}?top=10&after=${next}`;
+    const { data } = await axios.get(url, { headers: { Authorization: `Bearer ${token}` } });
+
+    configurations.push(...data.results);
+
+    if (data.page?.next) {
+      url = `/configuration/form-service?top=100&after=${data.page.next}`;
```
Comment (rstyan) [inline apps/tenant-management-webapp/src/app/store/configuration/sagas.ts:120]: Why is this hard coded to the form service?  All our services that have paging have the same interface, so why not use ${urlData.url}?top=...

File: apps/tenant-management-webapp/src/app/store/configuration/model.ts [hunk 6]
```diff
@@ -1,5 +1,12 @@
 import { RegisterData } from '@abgov/jsonforms-components';
 
+export const allNames = 'all names';
+
+export interface UrlData {
+  url?: string;
+  namespace?: boolean;
```
Comment (rstyan) [inline apps/tenant-management-webapp/src/app/store/configuration/model.ts:7]: I'm not convinced this boolean is needed.  See comments in the sagas.ts file.

**Review by rstyan — CHANGES_REQUESTED** [review]

**Pull request merged**